app/controllers/api/v1/shops_controller.rb

```diff
@@ -1,4 +1,7 @@
 class Api::V1::ShopsController < ApiController 
+  before_action :authenticate_user, except: [:index, :show]
+  before_action :authorize_user, except: [:index, :show, :create]
```

Comment:
we don't need to duplicate these methods to be used in both index and show - both methods check `if !user_signed_in?`, just the `authorize` also checks if the current user is an admin. Looks like `authorize` should be used for `create`, then index and show is just concerned with a user being signed in, thus using `authenticate`